libfwupdplugin/fu-self-test.c

```diff
@@ -1287,6 +1287,34 @@ fu_common_strstrip_func (void)
 	}
 }
 
+static void
+fu_common_version_semver_func (void)
+{
+	struct {
+		const gchar *old;
+		const gchar *new;
+	} map[] = {
+		{ "1.2.3",		"1.2.3" },
+		{ "1.2-3",		"1.2.3" },
+		{ "1~2-3",		"1.2.3" },
+		{ ".1.2",		"1.2" },
+		{ "1.2.",		"1.2" },
+		{ "1..2",		"1.2" },
+		{ "CBET1.2.3",		"1.2.3" },
+		{ "1.2.3alpha",		"1.2.3" },
+		{ "5",			"5" },
+		{ "\t5\n",		"5" },
+		{ "coreboot-unknown",	NULL },
+		{ "",			NULL },
+		{ " ",			NULL },
+		{ NULL,			NULL }
+	};
```

Comment:
@superm1 any more edge cases you can think of?

Reply:
Hex?

Reply:
Good idea, thanks.